packages/@sanity/form-builder/src/sanity/DefaultAssetSource/AssetUsageDialog.tsx

```diff
@@ -1,102 +1,113 @@
-  const footer = (
+export function AssetUsageDialog({
+  asset,
+  assetType = 'image',
+  mode = MODE_LIST_USAGE,
+  isDeleting = false,
+  onClose,
+  onDelete,
+}: UsageDialogProps) {
+  const isListMode = mode === MODE_LIST_USAGE
+  const defaultHeaderTitle = isListMode ? `Documents using ${assetType}` : `Delete ${assetType}`
+  const [canDelete, setCanDelete] = useState(false)
+  const [isLoadingParent, setIsLoadingParent] = useState(true)
+  const showActionFooter = mode === MODE_CONFIRM_DELETE
+  const footer = showActionFooter ? (
     <Grid padding={2} gap={2} columns={2}>
-      <Button text="Close" onClick={onClose} />
-      <Button text="Delete" tone="critical" icon={TrashIcon} onClick={onDelete} />
+      <Button mode="bleed" text="Cancel" onClick={onClose} />
+      <Button
+        text="Delete"
+        tone="critical"
+        icon={TrashIcon}
+        onClick={onDelete}
+        loading={isDeleting}
+        disabled={!canDelete}
+      />
     </Grid>
-  )
+  ) : undefined
+  const handleToggleCanDelete = ({
+    hasResults,
+    isLoading,
+  }: {
+    hasResults: boolean
+    isLoading: boolean
+  }) => {
+    setCanDelete(!hasResults && !isLoading)
+    setIsLoadingParent(isLoading)
+  }
 
   return (
     <Dialog
       id="asset-dialog"
-      header="Find usages of image"
-      width={2}
+      header={defaultHeaderTitle}
+      width={1}
       onClose={onClose}
       footer={footer}
+      __unstable_autoFocus={!isLoadingParent}
     >
-      <Box padding={4}>
-        <Grid gap={2} style={{gridTemplateColumns: 'max-content 1fr'}}>
-          <img
-            src={`${asset.url}?w=200`}
-            style={{maxWidth: '200px'}}
-            alt="The image used by the listed documents"
-          />
-
-          <WithReferringDocuments id={asset._id}>
-            {({isLoading, referringDocuments}) => {
-              const drafts = referringDocuments.reduce(
-                (acc, doc) => (doc._id.startsWith('drafts.') ? acc.concat(doc._id.slice(7)) : acc),
-                []
-              )
+      <WithReferringDocuments id={asset._id}>
+        {({isLoading, referringDocuments}) => {
+          const drafts = referringDocuments.reduce(
+            (acc, doc) => (doc._id.startsWith('drafts.') ? acc.concat(doc._id.slice(7)) : acc),
+            []
+          )
 
-              const filteredDocuments = referringDocuments.filter(
-                (doc) => !drafts.includes(doc._id)
-              )
+          const filteredDocuments = referringDocuments.filter((doc) => !drafts.includes(doc._id))
+          const hasResults = filteredDocuments.length > 0
+          const showDocumentList = mode === MODE_LIST_USAGE || hasResults
+          const noPaddingOnStack = mode === MODE_CONFIRM_DELETE && !hasResults
 
-              if (isLoading) {
-                return <SpinnerWithText text="Loading..." />
-              }
+          handleToggleCanDelete({hasResults, isLoading})
```

Comment:
I suspect this will make React issue a [warning](https://reactjs.org/blog/2020/02/26/react-v16.13.0.html#warnings-for-some-updates-during-render) about updating this component from inside the function body of a different component? This can be avoided by wrapping it in a <WithReferringDocuments> and pass the child props to it, something along the lines of:

```jsx
export function AssetUsageDialog(props) {
  return (
    <WithReferringDocuments id={props.asset._id}>
       {({isLoading, referringDocuments}) =>
         <InnerAssetUsageDialog
          {...props}
           assetIsLoading={isLoading}
           referringDocuments={referringDocuments}
       />
    </WithReferringDocuments>
  )
}
```

Reply:
Correct, I was trying to avoid wrapping it in another component, but I guess there is no way around it